boards/shields/nrf21540_ek/doc/index.rst

```diff
@@ -0,0 +1,84 @@
+.. _nrf21540_ek:
+
+nRF21540 EK
+###########
+
+The nRF21540 EK (Evaluation Kit) is an RF front-end module (FEM) for
+Bluetooth Low Energy, Bluetooth mesh, 2.4 GHz proprietary, Thread, and
+Zigbee range extension.
+When combined with an nRF52 or nRF53 Series SoC, the nRF21540 RF FEM’s
++21 dBm TX output power and 13 dB RX gain ensure a superior link budget for
+up to 16x range extension.
+
+The nRF21540 complementary device has a 50 Ω SMA input and 2x 50 Ω SMA outputs.
+This enables connecting an SoC or a signal generator to the input.
+It also enables connecting the outputs to measurement tools or to antennas
+directly.
+The FEM can be configured through the pins available on the Arduino headers.
+
+The nRF21540's gain control, antenna switching, and modes are controlled via
+GPIO or SPI, or a combination of both.
```

Comment:
If I am not mistaken, the shield will always enable the SPI. I think it is worth mentioning here, that the peripheral device will be occupied, even if the GPIO-only interface is used

Reply:
Yes it enables SPI always but maybe I will remove Kconfig.defconfig file which enables SPI and let user to decide whatever he want to use SPI also. What do you think @hubertmis ?

Reply:
I think it would be better. SPI should be enabled depending on selected nRF21540 driver. If given driver does not use SPI interface, there is no point in enabling SPI.

Reply:
Done

Reply:
Works for me!
What about the device tree overlay and the `spi-if` property?
Isn't it that way that if you set `spi-if = <&nrf_radio_fem_spi>;`, the driver will try to load SPI during the initialization?

Reply:
It doesn't load driver because default value for CONFIG_SPI is n so drivers won't be added into build tree. This only creates a DTS macros which can be used in your application

Reply:
Please note that the nrf21540dk board definition also added this in DTS

Reply:
Not necessarily. It is possible to have physical connection between nRF5 and nRF21540 SPI port and select nRF21540 GPIO driver. In such scenario, devicetree shall describe both GPIO and SPI connection, but the driver shall put SPI pins (at least CSN) to a known safe state. And this driver shall not enable SPI subsystem in Zephyr.

Reply:
So it should work in the following way:

Range of available nRF21540 drivers depends on entries in devicetree (GPIO, SPI, both, mixed, ...).
Driver's Kconfig should manage its dependencies (SPI, GPIO, whatever).

There should be no direct connection between nRF21540 device tree and driver dependencies, like "if you have SPI device, enable SPI driver".

Reply:
Thank you for the clarification!